packages/trace-viewer/src/ui/uiModeView.tsx

```diff
@@ -247,20 +254,25 @@ export const UIModeView: React.FC<{}> = ({
     testTree.sortAndPropagateStatus();
     testTree.shortenRoot();
     testTree.flattenForSingleProject();
-    setVisibleTestIds(testTree.testIds());
     return { testTree };
-  }, [filterText, testModel, statusFilters, projectFilters, setVisibleTestIds, runningState, isRunningTest, mergeFiles]);
+  }, [filterText, testModel, statusFilters, projectFilters, runningState, isRunningTest, mergeFiles]);
 
-  const runTests = React.useCallback((mode: 'queue-if-busy' | 'bounce-if-busy', testIds: Set<string>) => {
+  const runTests = React.useCallback((mode: 'queue-if-busy' | 'bounce-if-busy', testTree: TestTree, testItems: Iterable<TreeItem>) => {
```

Comment:
Can `testTree` differ from the one declared at line 249? I think we should not pass it.

Reply:
done